anime_downloader/sites/yify.py

```diff
@@ -0,0 +1,45 @@
+import requests, json, os
+from anime_downloader.sites.anime import Anime, AnimeEpisode, SearchResult
+from anime_downloader.sites import helpers
+from bs4 import BeautifulSoup
+from io import StringIO
+
+class Yify(Anime, sitename = 'yify'):
+    sitename = 'yify'
+    url = f'https://{sitename}.mx'
+    @classmethod
+    def search(cls, query):
+        search_results = requests.get(f'https://yts.mx/ajax/search?query={query}').text
+        io = StringIO(search_results)
+        results = json.load(io)
+        results_list = []
+        for movie in results['data']:
+            
+            """
+            TODO: Find a way to make quality selesction possible.
+                  720p is 'modal-quality-720p'
+                  1080p is 'modal-quality-1080p'
+                  For now the default value is 720p.
+            """
+            html = BeautifulSoup(requests.get(movie['url']).text, 'html.parser')
```

Comment:
This is wrong. You should return `SearchResult`s here and scrape the data in `_scrape_episodes`